tests/io/test_iris.py

```diff
@@ -197,3 +198,50 @@ def test_decode_string():
 def test__get_fmt_string():
     fmt = "<12sHHi12s12s12s6s12s12sHiiiiiiiiii2sH12sHB1shhiihh80s16s12s48s"
     assert iris._get_fmt_string(iris.PRODUCT_CONFIGURATION) == fmt
+
+
+def test_read_from_record(iris0_file, file_or_filelike):
+    with _get_data_file(iris0_file, file_or_filelike) as testfile:
+        try:
+            ingest_data_file = IrisIngestDataFile(filename=testfile, loaddata=False)
+            result = ingest_data_file.read_from_file(10)  # Modify `10` as needed
+            assert isinstance(result, bytes)
+        except OSError as e:
+            assert "Cannot read" in str(e)
+
+
+def test_decode_data(iris0_file, file_or_filelike):
+    with _get_data_file(iris0_file, file_or_filelike) as testfile:
+        try:
+            ingest_data_file = IrisIngestDataFile(filename=testfile, loaddata=False)
+            sample_data = np.array([1, 2, 3, 4], dtype=np.uint16)
+            prod_dict = {
+                "func": None,  # Replace with actual function if applicable
+                "dtype": "uint16",
+            }
+            decoded_data = ingest_data_file.decode_data(sample_data, prod_dict)
+            assert isinstance(decoded_data, np.ndarray)
+        except OSError as e:
```

Comment:
@syedhamidali Can you shortly explain what the except-branch is doing?

Reply:
Oh, I see. So this iris0_file hasn't been able to load via pytests locally, I put exceptions, but I didn't see it is going to pass the tests anyway. :)

Reply:
Please use `iris0_file` fixture, what did not work locally?

Reply:
The problem might be, that `iris0_file` is no `IrisIngestDataFile` but `IrisRawFile`, no?

Reply:
Yes, let me do that instead. I see where the error is. Thanks!